clang/lib/AST/DeclPrinter.cpp

```diff
@@ -855,7 +868,17 @@ void DeclPrinter::VisitFunctionDecl(FunctionDecl *D) {
         Out << Proto << " -> ";
         Proto.clear();
       }
-      AFT->getReturnType().print(Out, Policy, Proto);
+      if (!Policy.SuppressTagKeyword && Policy.SuppressScope &&
+          !Policy.SuppressUnwrittenScope) {
+        AddPrefix(Policy, AFT->getReturnType(), Out);
```

Comment:
This check is duplicated in `AddPrefix()`, same below (probably can remove the check from `AddPrefix()` or turn it into an assertion).